src/BeanstalkClient.php

```diff
@@ -294,24 +318,59 @@ public function getSystemStats(): Promise {
 
             switch ($type) {
                 case "OK":
-                    return new System($this->getStatsFromString($response[1]));
+                    return new System(Yaml::parse($response[1]));
 
                 default:
                     throw new BeanstalkException("Unknown response: " . $type);
             }
         });
     }
 
-    private function getStatsFromString(string $stats): array {
-        $result = [];
-        $source = explode("\n", $stats);
-        foreach ($source as $stat) {
-            if ($stat == '---' || empty($stat)) {
-                continue;
+    public function listTubes(): Promise {
+        $payload = "list-tubes\r\n";
+
+        return $this->send($payload, function (array $response): array {
+            list($type) = $response;
+
+            switch ($type) {
+                case "OK":
+                    return Yaml::parse($response[1]);
+
+                default:
+                    throw new BeanstalkException("Unknown response: " . $type);
             }
-            list($key, $value) = explode(':', $stat);
-            $result[$key] = trim($value);
-        }
-        return $result;
+        });
+    }
+
+    public function listWatchedTubes(): Promise {
+        $payload = "list-tubes-watched\r\n";
+
+        return $this->send($payload, function (array $response): array {
+            list($type) = $response;
+
+            switch ($type) {
+                case "OK":
+                    return Yaml::parse($response[1]);
+
+                default:
+                    throw new BeanstalkException("Unknown response: " . $type);
+            }
+        });
+    }
+
+    public function getUsingTube(): Promise {
```

Comment:
Could you rename that one to `getUsedTube()`?

Reply:
Really? What name you want to see here ?

Reply:
`getUsedTube()`, as mentioned. :smile: 

Other than that it looks perfectly fine now. :+1: